pages/index.tsx

```diff
@@ -24,6 +24,13 @@ export function getServerSideProps(context) {
     }
   }
 
+
+  if(username) return {
+    redirect: {
+      destination: `/${username}+`
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new redirection logic based on `username` is introduced without any validation or sanitization of the `username`. This could potentially lead to Open Redirect vulnerabilities if the `username` parameter can be controlled by an attacker. Consider validating the `username` before using it in a redirect URL.

```diff
  if(username) return {
    redirect: {
+      // Validate username before redirecting
+      destination: isValidUsername(username) ? `/${username}+` : '/error'
    }
  }
```
In the above code, `isValidUsername()` is a hypothetical function that you would need to implement. This function should check if the `username` is valid according to your application's rules (e.g., allowed characters, length). If the `username` is not valid, the user is redirected to an error page.


<!-- This is an auto-generated comment by OSS CodeRabbit -->